src/main/java/org/openlmis/fulfillment/web/OrderController.java

```diff
@@ -192,15 +190,14 @@ public OrderDto createOrder(@RequestBody OrderDto orderDto,
   @ResponseStatus(HttpStatus.OK)
   @ResponseBody
   public Page<BasicOrderDto> searchOrders(OrderSearchParams params, Pageable pageable) {
-    List<Order> orders = orderService
-        .searchOrders(params)
-        .stream()
-        .filter(Objects::nonNull)
-        .sorted((o1, o2) -> ObjectUtils.compare(o1.getId(), o2.getId()))
-        .collect(Collectors.toList());
+    List<Order> orders = orderService.searchOrders(params);
 
     List<Order> filteredList = orderSecurityService.filterInaccessibleOrders(orders);
-    Iterable<BasicOrderDto> data = BasicOrderDto.newInstance(filteredList, exporter);
+    // with this we will only create basic order dto objects for elements that would be
+    // returned by this endpoint
+    Page<Order> page = Pagination.getPage(filteredList, pageable);
```

Comment:
Yes the best solution would be move pagination on the db. Currently the problem is that if we have for example 10,000 orders after permission check then there will be about 60,000 requests to reference-data. After that changes the list will be shorten to 10 orders so there will be 60 requests. We found small bug in those changes because totalElements fields return 10 which not always is correct.